src/scripts/ordo_report_mapping_annotations/create_sparql__ordo_add_annotation_based_mappings.py

```diff
@@ -0,0 +1,135 @@
+"""Update ORDO mapping annotations to formalize them semantically.
+Work described furthere here: https://github.com/monarch-initiative/mondo-ingest/issues/23
+
+Resources
+1. https://www.orpha.net/orphacom/cahiers/docs/GB/Orphanet_ICD10_coding_rules.pdf
+
+Prerequisites to run this script:
+1. `src/ontology/mirror/ordo.owl` must be present. To get it, run `sh src/ontology/run.sh make refresh-ordo`.
+2. Your pwd (present working directory) should be `src/scripts/ordo_report_mapping_annotations` in order for local imports to work.
+
+TODO's
+TODO 1: `robot` --> `sh run.sh robot...`
+TODO 2. Should delete any existing `src/sparql/update-mapping-annotations[0-9].*\.ru` files prior to creating new ones.
+ ...When this is done, the comment above the make goal in `src/ontology/mondo-ingest.Makefile` should be updated, as it
+ ...will no longer be necessary to do this manually.
+"""
+import os
+import subprocess
+from typing import Dict, List
+
+from jinja2 import Template
+
+from report_mapping_annotations import run as get_report
+
+
+# Vars
+# # Paths
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+CACHE_DIR = os.path.join(SCRIPT_DIR, 'cache')
+PROJECT_DIR = os.path.join(SCRIPT_DIR, '..', '..', '..')
+SOURCES_DIR = os.path.join(PROJECT_DIR, 'src', 'ontology', 'mirror')
+SPARQL_DIR = os.path.join(PROJECT_DIR, 'src', 'sparql')
+# # Messages
+REQS_ERR = '`{}` must be present. To get it, run `sh src/ontology/run.sh make refresh-ordo`.'
+# # Config
+CONFIG = {
+    'query_template_path': {
+        'ontology': os.path.join(SOURCES_DIR, 'ordo.owl'),
+        'sparql_query': os.path.join(SPARQL_DIR, 'ordo-add-annotation-based-mappings.ru.jinja2'),
+    },
+    'output_path': os.path.join(PROJECT_DIR, 'ordo-new.owl'),
+    'run_command': False  # If false, just instantiates templates and saves. If true, also runs them.
+    # 'use_cache': True  # saves instantiated strings to cache/ anyway, but doesn't re-use them atm
+}
+
+
+# Functions
+def sparql_jinja2_file_query__via_robot(
+    onto_path: str, query_template_path: str, output_path: str, mapping_str__curie__map: Dict, run_command=False
+) -> Dict[str, str]:
+    """Query ontology using SPARQL query file
+
+    :param: run_command (bool): If True, then in addition to instantiating the Jinja template, it will also run the
+    instantiated queries.
+    """
+    results_dirname = os.path.basename(onto_path).replace('/', '-').replace('.', '-') + \
+        '__' + os.path.basename(query_template_path).replace('.', '-')
+    results_dirpath = os.path.join(CACHE_DIR, 'robot', results_dirname)
```

Comment:
@matentzn In my other `create_sparql_*.py` file, I left out the caching bit. I have intentionally introduced some variation in how these Python files are set up, so that you can review the differences and see which kind of setup you like best.